File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/matchstate/GameState.java
package org.akoshterek.backgammon.matchstate;

/**
 * @author Alex
 *         date 06.08.2015.
 */
public enum GameState {
    GAME_NONE(0),
    GAME_PLAYING(1),
    GAME_OVER(2);

    private final int value;

    GameState(final int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/matchstate/GameState.scala
package org.akoshterek.backgammon.matchstate

/**
  * Created by Alex on 29-04-17.
  */
sealed trait GameState {
    val value: Int
}

object GameState {
    case object GAME_NONE extends GameState { val value = 0 }
    case object GAME_PLAYING extends GameState { val value = 1 }
    case object GAME_OVER extends GameState { val value = 2 }
}